Assets/Scripts/Achievement/AchievementManager.cs

```diff
@@ -13,73 +15,114 @@ public class AchievementManager : Singleton<AchievementManager>
         [SerializeField]
         private AchievementScriptable achievementScriptable;
 
-        public AchievementController hiScoreController { get; private set; }
-        public AchievementController gamesPlayedController { get; private set; }
-        public AchievementController enemiesKilledController { get; private set; }
+        private List<Achievement> achievementList;
 
-        public event Action<AchievementType, int> AchievementCheck;
-        public event Action<string> AchievementUI;
+        public List<Achievement> AchievementList { get { return achievementList; }}
 
-        // Use this for initialization
-        void Start()
-        {
-            GameUI.InstanceClass.ScoreIncreased += ScoreIncreased;
-            Enemy.EnemyManager.Instance.EnemyDestroyed += EnemyKilled;
-            BTManager.GameManager.Instance.GameStarted += GamesPlayed;
-            SaveLoad.SaveLoadManager.Instance.AchievementUnlocked += CallAchievementUI;
+        public event Action<int> AchievementCheck;
+        public event Action<string> AchievementUnlocked;
 
+        bool initialized = false;
 
-            if (achievementScriptable != null)
-            {
-                hiScoreController       = new AchievementController(achievementScriptable, AchievementType.hiScore);
-                gamesPlayedController   = new AchievementController(achievementScriptable, AchievementType.gamesPlayed);
-                enemiesKilledController = new AchievementController(achievementScriptable, AchievementType.enemyKilled);
-            }
-            else 
+        public void AchievmentInitialize()
+        {
+            if (initialized == false)
             {
-                Debug.Log("[Achievement Manager] missing scriptable object reference");
+                initialized = true;
+                BTManager.GameManager.Instance.GameStarted += InvokeDefaultEvents;
+                if (achievementScriptable != null)
+                {
+                    achievementList = new List<Achievement>();
+
+                    for (int i = 0; i < achievementScriptable.achievementList.Count; i++)
+                    {
+                        Achievement achievement = new Achievement();
+                        achievement = achievementScriptable.achievementList[i];
+                        string dataString = "Achievement_" + i;
+                        bool unlocked = SaveLoadManager.Instance.GetAchievementProgress(dataString, i);
+                        if (unlocked == true)
+                            achievement.achievementInfo.achievementStatus = AchievementStatus.Unlocked;
+
+                        achievementList.Add(achievement);
+                    }
+                }
+                else
+                {
+                    Debug.Log("[Achievement Manager] missing scriptable object reference");
+                }
             }
+        }
 
+        private void InvokeDefaultEvents()
```

Comment:
you are subscribing or registering to events not invoking them - name is misleading

Reply:
Fixed